Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TriangularFEMForceField.inl

```diff
@@ -1157,10 +1157,41 @@ void TriangularFEMForceField<DataTypes>::addForce(const core::MechanicalParams*
     if (f_computePrincipalStress.getValue() || p_computeDrawInfo)
     {
         unsigned int nbTriangles = m_topology->getNbTriangles();
-        type::vector<TriangleInformation>& triangleInf = *(triangleInfo.beginEdit());
+        auto triangleInf = sofa::helper::getWriteOnlyAccessor(triangleInfo);
         for (unsigned int i = 0; i < nbTriangles; ++i)
             computePrincipalStress(i, triangleInf[i].stress);
-        triangleInfo.endEdit();
+
+
+        if (showStressValue.getValue()) // if true will compute averageStress per point
+        {
+            auto triangles = m_topology->getTriangles();
```

Comment:
```suggestion
           const auto& triangles = m_topology->getTriangles();
```

Reply:
I thought if a method was returning a const &,  auto would be automatically a const & as well. I'm wrong?

Reply:
hum I dont think the compiler could guess that you wont touch the triangles afterwards...